language/ir-testsuite/tests/scripts/script_with_type_parameters.mvir

```diff
@@ -0,0 +1,4 @@
+//! type-args: u64, u8, 0x0::LibraAccount::T
+main<T1, T2: unrestricted, T3: resource>() {
+    return;
+}
```

Comment:
Could you add some tests to ensure that `main<T>(t: T)`, `main<T>(t: vector<T>)` and other ways of using the type parameters in the value parameters are rejected?

Reply:
Thanks for catching this!